.github/workflows/build-and-test.yml

```diff
@@ -487,6 +487,63 @@ jobs:
             throw "Failed to uninstall fluentdwinsvc."
           }
 
+  windows-zeroconfig-e2e-test:
```

Comment:
I'm not sure this needs to run w/ `build-and-test` and think a new zero config workflow could be added that is limited to `internal/buildscripts/packaging/msi` and `instrumentation` content changes.

Reply:
Only downside to moving to a new workflow is that it would require duplicating the build for all upstream deps (otelcol, agent bundle, and msi).

Alternatively, maybe we can try something like https://github.com/marketplace/actions/paths-changes-filter to only run the job when there are relevant changes?

Reply:
If adding the 3rd party action is acceptable it sounds good to me - this action seems reputable. Related Q.: is this enough to guarantee that it runs at least once for each release?

Reply:
We could also run this for merges to main for general tracking coverage?

Reply:
If we want to use that action, I think we can add the action to the upstream `windows-msi` job and configure this job to always run for `main` or for relevant changes.  Maybe something like:
```
windows-msi:
  outputs:
    zeroconfig: ${{ steps.filter.outputs.zeroconfig }}
  steps:
    - name: Check out the codebase.
      uses: actions/checkout@v3
    - uses: dorny/paths-filter@v2
      id: filter
      with:
        filters: |
          zeroconfig:
            - path1
            - path2
            - ...
```
and then
```
windows-zeroconfig-e2e-test:
  needs: [ windows-msi ]
  if: ${{ github.ref_name == 'main' || needs.windows-msi.outputs.zeroconfig == 'true' }}
  steps: ...
```

Reply:
I guess this is the last pending thing to merge this PR. I will take a look in the afternoon PT.

Reply:
I opted to have a dedicated job instead of piggybacking on `windows-msi` since the job is very light weight, but, no strong opinion about that, see https://github.com/signalfx/splunk-otel-collector/pull/2901/commits/2ce7fb89dca76f1569d7968a2857d99b9877eeda